File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/http/ServiceStarter.java
package ch.rmy.android.http_shortcuts.http;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;

import ch.rmy.android.http_shortcuts.utils.Connectivity;

public class ServiceStarter extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (!ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
            return;
        }
        if (!Connectivity.isNetworkConnected(context)) {
            return;
        }
        context.startService(new Intent(context, ExecutionService.class));
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/http/ServiceStarter.kt
package ch.rmy.android.http_shortcuts.http

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager

import ch.rmy.android.http_shortcuts.utils.Connectivity

class ServiceStarter : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (ConnectivityManager.CONNECTIVITY_ACTION != intent.action || !Connectivity.isNetworkConnected(context)) {
            return
        }
        context.startService(Intent(context, ExecutionService::class.java))
    }

}
